diffkemp/snapshot.py

```diff
@@ -186,9 +199,17 @@ def _from_yaml(self, yaml_file):
         self.created_time = self.created_time.replace(
             tzinfo=datetime.timezone.utc)
 
-        if os.path.isdir(yaml_dict["source_kernel_dir"]):
-            self.kernel_source = KernelSource(yaml_dict["source_kernel_dir"],
-                                              True)
+        llvm_finder_cls = None
+        llvm_finder_path = None
+        if yaml_dict["llvm_source_finder"]["kind"] == "kernel_with_builder":
+            llvm_finder_cls = KernelLlvmSourceBuilder
+        if llvm_finder_cls is not None:
+            self.snapshot_tree.init_source_finder(llvm_finder_cls,
+                                                  llvm_finder_path)
+
+        if os.path.isdir(yaml_dict["source_dir"]):
```

Comment:
Here it might make sense to print a warning if the source directory doesn't exist, since that means some functionality of DiffKemp will be limited (e.g. C function call argument equality).